wagtail/admin/tests/pages/test_reorder_page.py

```diff
@@ -140,3 +178,75 @@ def setUp(self):
 
         # Login
         self.user = self.login()
+
+
+class TestPageReorderWithNoSubPageTypes(TestPageReorder):
+    """
+    This TestCase is the same as the TestPageReorder class above, but uses a
+    NoSubpageTypesPage instance as the parent page instead of SimplePage.
+
+    This ensures that a parent page having an empty `subpage_types` list does
+    not prevent the reorder pre-existing or automatically generated pages
+    beneath it.
```

Comment:
@ababic Thanks, sorry I didn't get to this before the 6.2 feature freeze.

The `PermissionDenied` was raised by the `MovePageAction` class which is called when the `move()` is done:

https://github.com/wagtail/wagtail/blob/9f518a007d73930c04d5c7e810c1f475190c82f6/wagtail/admin/views/pages/ordering.py#L32-L42

https://github.com/wagtail/wagtail/blob/9f518a007d73930c04d5c7e810c1f475190c82f6/wagtail/models/__init__.py#L2371-L2376

https://github.com/wagtail/wagtail/blob/9f518a007d73930c04d5c7e810c1f475190c82f6/wagtail/actions/move_page.py#L28-L33

The reason is because `PagePermissionTester.can_move_to(destination)` checks for `Page.can_move_to(destination)`, which in turn also checks for `Page.can_exist_under(parent)` that returns `False` when the moved page type is not in the parent's `subpage_types`.

While I initially considered this as an edge case that's not worth considering (https://github.com/wagtail/wagtail/pull/11944#pullrequestreview-2063002247), after reading this test's docstring, I think this is worth fixing too.

I pushed a commit [`7196066` (#11944)](https://github.com/wagtail/wagtail/pull/11944/commits/71960666691f921fecf80651a317b8e68a1c9b95) that should fix this. Do you think that's a reasonable fix? If so, I'll rebase and clean up the PR to be merged for 6.3. Thanks!

Reply:
Yes, short of making `can_exist_under()` an instance method and handling this there (which would no doubt introduce further problems), your change appears to do the job nicely! It is a shame to introduce multiple meanings of 'can exist under'.

Reply:
Thinking some more, how would you feel about adding a separate `can_exist_under_page()` method, put it right next to `can_exist_under()` and use docstrings to outline the difference? This would make the API much clearer overall, I think.

Reply:
Yeah, I wanted to add it inside `can_exist_under()` but unfortunately it's a classmethod. I'm not sure about adding a `can_exist_under_page()`, from the naming perspective it reads like the `_page` is referring to the parent page, which wouldn't be any different to the existing classmethod that also takes the parent page instance as a parameter. If we were to add a new method, maybe `page_can_exist_under()` helps avoid that confusion – but honestly I don't even understand why we have both `Page.can_move_to()` and `PagePermissionTester.can_move_to()`, it seems like the former's logic should've been implemented into the latter.

I opted to put the logic into `PagePermissionTester.can_move_to()` instead, and opened a new rebased and cleaned PR here, so others can review: [`534c5e4` (#12156)](https://github.com/wagtail/wagtail/pull/12156/commits/534c5e4b437e50a6383da0468f44e6ab0c8a15a5)